### PR TITLE
Ravtdbq 344 support for shiftright function

### PR DESCRIPTION
**Pull request by Abhi-210793**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 1]
```diff
@@ -1239,10 +1242,30 @@ private void unParseRegexpLike(SqlWriter writer, SqlCall call, int leftPrec, int
     writer.endFunCall(ifFrame);
   }
 
-  private void unparseShiftLeft(SqlWriter writer, SqlCall call) {
+  private void unparseShiftLeftAndShiftRight(SqlWriter writer, SqlCall call, boolean isShiftLeft) {
```
Comment (sauajmera) [inline core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:1245]: can we refactor this code and extract some code to function to make it more clear.
I have create 3 methods and separate out the code to make it more clear.
1) isBasicCallWithNegativePrefix
2) getPositiveOperand
3) getShiftOperator
eg : 


private void unparseShiftLeftAndShiftRight(SqlWriter writer, SqlCall call, boolean isShiftLeft) {
    call.operand(0).unparse(writer, 0, 0);
    SqlNode secondOperand = call.operand(1);

    if (isBasicCallWithNegativePrefix(secondOperand)) {
        SqlNode positiveOperand = getPositiveOperand(secondOperand);
        writer.print(getShiftOperator(isShiftLeft));
        writer.print(" ");
        positiveOperand.unparse(writer, 0, 0);
    } else {
        writer.print(getShiftOperator(isShiftLeft));
        writer.print(" ");
        secondOperand.unparse(writer, 0, 0);
    }
}


private boolean isBasicCallWithNegativePrefix(SqlNode secondOperand) {
    return secondOperand instanceof SqlBasicCall &&
            ((SqlBasicCall) secondOperand).getOperator().getKind() == SqlKind.MINUS_PREFIX;
}

private SqlNode getPositiveOperand(SqlNode secondOperand) {
    return ((SqlNode[]) ((SqlBasicCall) secondOperand).operands)[0];
}

private String getShiftOperator(boolean isShiftLeft) {
    return isShiftLeft ? SHIFTLEFT : SHIFTRIGHT;
}

Comment (sauajmera) [reply core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:1245]: also check if ((SqlNode[]) ((SqlBasicCall) secondOperand).operands)[0]; condition can be replaced by ((SqlBasicCall) firstOperand).operand(0)

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 2]
```diff
@@ -1239,10 +1242,30 @@ private void unParseRegexpLike(SqlWriter writer, SqlCall call, int leftPrec, int
...
     call.operand(0).unparse(writer, 0, 0);
-    writer.print(SHIFTLEFT + " ");
-    call.operand(1).unparse(writer, 0, 0);
+    SqlNode firstOperand = call.operand(1);
+    if (firstOperand instanceof SqlBasicCall
+        && (((SqlBasicCall) firstOperand).getOperator()).kind == SqlKind.MINUS_PREFIX) {
+      // If the second operand is negative, fetch the postive value and change the operator
```
Comment (sauajmera) [inline core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:1250]: check "positive" spelling in comment and also if we create new methods as suggested then these comments can be put on the top of new methods.

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java (hunk 1 above)
Comment (Abhi-210793) [reply core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:1245]: Done

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java (hunk 2 above)
Comment (Abhi-210793) [reply core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java]: Done

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 3]
```diff
@@ -1224,6 +1224,9 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
     case "BITNOT":
       unparseBitNotFunction(writer, call);
       break;
+    case "SHIFTRIGHT":
+      unparseShiftRight(writer, call);
```
Comment (sauajmera) [inline core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:1228]: can we remove these two functions unparseShiftRight(writer, call) & unparseShiftLeft(writer, call) and can directly use
unparseShiftLeftAndShiftRight(writer, call, true/false) calls respectively.

Comment (Abhi-210793) [reply core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java]: done

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java (hunk 1 above)
Comment (sauajmera) [reply core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:1245]: changes looks good.

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java (hunk 3 above)
Comment (sauajmera) [reply core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java]: looks good.

**Review by sauajmera — APPROVED** [review]

**Pull request merged**